k-distribution/tests/regression-new/map-symbolic-tests/assignment-11-spec.k.out

```diff
@@ -0,0 +1,4 @@
+<k>
+  assignmentResult ( Y |-> 2
+  ( MAP [ Z <- 3 ] ) )
```

Comment:
the ideal output should be: `Y |-> 2 Z |-> 3 MAP` (or one with replacing Y with X).
the current output is not fully evaluated.
there would be many partially evaluated outputs.
so, your plan is to update this output if the next version changes the behavior and returns another form of the output?

Reply:
This seems to imply that `MAP [ Z <- 3 ] => Z |-> 3 MAP` should be provable, but I don't think it should given the precondition?

Reply:
Added tests with both `Z in_keys(MAP)` and `notBool Z in_keys(MAP)`, so we test both sides.